tasks/go_test.py

```diff
@@ -895,3 +896,67 @@ def format_packages(ctx, impacted_packages):
 
 def normpath(path):  # Normpath with forward slashes to avoid issues on Windows
     return os.path.normpath(path).replace("\\", "/")
+
+
+@task
+def notify_unit_tests(ctx, pipeline_id, pipeline_url, branch_name):
+    from .libs.common.github_api import GithubAPI
+
+    pipeline_id_regex = re.compile(r"pipeline ([0-9]*)")
+
+    no_tests_executed_jobs = process_unit_tests_tarballs(ctx)
+    gh = GithubAPI("DataDog/datadog-agent")
+    pr = gh.get_pr_for_branch(branch_name)[0]
+    # If the branch is not linked to any PR we stop here
+    if pr is None:
+        return
+
+    comment = gh.find_comment(pr.number, "[Fast Unit Tests Report]")
+    if comment is None and len(no_tests_executed_jobs) > 0:
+        msg = create_msg(pipeline_id, pipeline_url, no_tests_executed_jobs)
+        gh.publish_comment(pr.number, msg)
+        return
+
+    previous_comment_pipeline_id = pipeline_id_regex.findall(comment.body)[0]
+    # An older pipeline should not edit a message corresponding to a newer pipeline
+    if previous_comment_pipeline_id > pipeline_id:
+        return
+
+    if len(no_tests_executed_jobs) > 0:
+        msg = create_msg(pipeline_id, pipeline_url, no_tests_executed_jobs)
+        comment.edit(msg)
+    else:
+        comment.delete()
+
+
+def create_msg(pipeline_id, pipeline_url, job_list):
+    msg = f'''
+[Fast Unit Tests Report]
+
+Warning: On pipeline [{pipeline_id}]({pipeline_url}). The following jobs did not run any unit tests:
+'''
+    for job in job_list:
+        msg += f"  - {job}\n"
+    msg += "\n"
+    msg += "If you modified Go files and expected unit tests to run in these jobs, please double check the job logs. If you think tests should have been executed reach out to #agent-platform"
+
+    return msg
+
+
+def process_unit_tests_tarballs(ctx):
+    tarballs = ctx.run("ls junit-tests_*.tgz", hide=True).stdout.split()
+    no_tests_executed_jobs = []
+    for tarball in tarballs:
+        ctx.run("mkdir -p extract_folder")
+        ctx.run(f"tar -xzf {tarball} -C extract_folder")
+
+        # We check if the folder contains at least one junit.xml file if not we consider no tests were executed
+        junit_files = ctx.run("ls extract_folder/*.xml", hide=True, warn=True)
+        if junit_files.exited != 0:
```

Comment:
```suggestion
        if not any(f.endswith(".xml") for f in os.listdir(unpack_dir)):
```
maybe a little more elegant rather than checking the exit value of a `ls`, wdyt?